From b95f55365f3b221be032b1008bb8032856e31a12 Mon Sep 17 00:00:00 2001
From: Jacek Tomasiak <jacek.tomasiak@gmail.com>
Date: Tue, 15 Mar 2022 23:57:12 +0100
Subject: [PATCH] Add branding for SLE Micro

This change requires distribution-logos-SLE and wallpaper-branding-SLE
to be available during build and installed on the target system to work
properly.
---
 Makefile.am                            |  1 +
 src/branding/sle-micro/Makefile.am  | 11 ++++
 src/branding/sle-micro/branding.css | 87 ++++++++++++++++++++++++++
 3 files changed, 99 insertions(+)
 create mode 100644 src/branding/sle-micro/Makefile.am
 create mode 100644 src/branding/sle-micro/branding.css

Index: cockpit-309/Makefile.am
===================================================================
--- cockpit-309.orig/Makefile.am
+++ cockpit-309/Makefile.am
@@ -175,6 +175,7 @@ include src/branding/default/Makefile.am
 include src/branding/fedora/Makefile.am
 include src/branding/kubernetes/Makefile.am
 include src/branding/opensuse/Makefile.am
+include src/branding/sle-micro/Makefile.am
 include src/branding/registry/Makefile.am
 include src/branding/rhel/Makefile.am
 include src/branding/scientific/Makefile.am
Index: cockpit-309/src/branding/sle-micro/Makefile.am
===================================================================
--- /dev/null
+++ cockpit-309/src/branding/sle-micro/Makefile.am
@@ -0,0 +1,11 @@
+slemicrobrandingdir = $(datadir)/cockpit/branding/sle-micro
+
+dist_slemicrobranding_DATA = \
+	src/branding/sle-micro/branding.css \
+	$(NULL)
+
+install-data-hook::
+	ln -sf /usr/share/wallpapers/SLEdefault/contents/images/1920x1200.png $(DESTDIR)$(slemicrobrandingdir)/default-1920x1200.png
+	ln -sf /usr/share/pixmaps/distribution-logos/square-hicolor.svg $(DESTDIR)$(slemicrobrandingdir)/square-hicolor.svg
+	ln -sf /usr/share/pixmaps/distribution-logos/favicon.ico $(DESTDIR)$(slemicrobrandingdir)/favicon.ico
+	ln -sf /usr/share/pixmaps/distribution-logos/apple-touch-icon.png $(DESTDIR)$(slemicrobrandingdir)/apple-touch-icon.png
Index: cockpit-309/src/branding/sle-micro/branding.css
===================================================================
--- /dev/null
+++ cockpit-309/src/branding/sle-micro/branding.css
@@ -0,0 +1,87 @@
+/* Extra overrides */
+:root {
+    --eos-bc-green-500: #30ba78;
+    --eos-bc-pine-500: #0c322c;
+    --eos-bc-gray-50: #F2F2F2;
+    --eos-bc-white: #ffffff;
+    --eos-bc-text: #333;
+}
+
+
+#brand:before {
+    content: "${PRETTY_NAME}";
+}
+
+.login-pf .container {
+    background-color: rgba(255, 255, 255);
+}
+
+#option-group svg polygon {
+    fill: var(--eos-bc-text);
+}
+
+/* Nav & sidebar overwrites */
+.navbar-pf {
+   background: var(--eos-bc-pine-500);
+}
+
+.navbar-pf-vertical {
+  border-top: none;
+}
+
+
+/* Default overrides */
+
+.login-note {
+    color: transparent;
+    position: relative;
+}
+
+body.login-pf {
+    background-color: var(--eos-bc-pine-500);
+    color: var(--eos-bc-text);
+}
+
+/* Only use background image on desktops */
+@media(min-width: 1024px) {
+    body.login-pf {
+        background-image: url("default-1920x1200.png");
+        background-repeat: no-repeat;
+        background-position: 100% 0 !important;
+        background-size: cover;
+    }
+}
+/* Hide badge on desktops (already included in background) */
+@media(max-width: 1023px) {
+    #badge {
+        width: 51px;
+        height: 45px;
+        background-image: url("square-hicolor.svg");
+        background-repeat: no-repeat;
+    }
+}
+
+#brand {
+    width: auto;
+    height: auto;
+    background-image: none;
+    background-repeat: no-repeat;
+    font-size: 2em;
+    white-space: nowrap;
+}
+/* Switch to small font faster to not clip long name */
+@media(max-width: 560px) {
+    #brand {
+        font-size: inherit;
+    }
+}
+
+#index-brand {
+    font-weight: bold;
+}
+
+#index-brand:before {
+    content: "${PRETTY_NAME}";
+}
+
+
